frontend/src/component/pages/shared/UserShortLinksSection.spec.tsx

```diff
@@ -0,0 +1,190 @@
+        onPageLoad={jest.fn}
+        pagedShortLinks={pagedShortLinks}
+      />
+    );
+
+    expect(container.textContent).toContain('Created Short Links');
+    expect(container.textContent).toContain('Long URL');
+    expect(container.textContent).toContain('Alias');
+  });
+
+  test('should render passed short links correctly', () => {
+    const pagedShortLinks: IPagedShortLinks = {
+      shortLinks: sampleShortLinks.slice(0, 3),
+      totalCount: sampleShortLinks.length
+    };
+
+    const { container } = render(
+      <UserShortLinksSection
+        onPageLoad={jest.fn}
+        pagedShortLinks={pagedShortLinks}
+      />
+    );
+
+    expect(container.textContent).toContain('Created Short Links');
+    expect(container.textContent).toContain('Long URL');
+    expect(container.textContent).toContain('Alias');
+
+    for (let urlIdx = 0; urlIdx < pagedShortLinks.shortLinks.length; urlIdx++) {
+      const shortLink = pagedShortLinks.shortLinks[urlIdx];
+      expect(container.textContent).toContain(shortLink.originalUrl);
+      expect(container.textContent).toContain(shortLink.alias);
+    }
+  });
+
+  test('should render correctly when total short links passed is less than page size', () => {
+    const pagedShortLinks: IPagedShortLinks = {
+      shortLinks: sampleShortLinks.slice(0, 3),
+      totalCount: 3
+    };
+    const pageSize = 5;
+
+    const { container } = render(
+      <UserShortLinksSection
+        onPageLoad={jest.fn}
+        pagedShortLinks={pagedShortLinks}
+        pageSize={pageSize}
+      />
+    );
+
+    expect(container.querySelectorAll('.page-number')).toHaveLength(1);
+  });
+
+  test('should render multiple pages when total short link count exceeds page size', () => {
+    const pageSize = 5;
+    const pagedShortLinks: IPagedShortLinks = {
+      shortLinks: sampleShortLinks.slice(0, pageSize),
+      totalCount: sampleShortLinks.length
+    };
+
+    const { container } = render(
+      <UserShortLinksSection
+        onPageLoad={jest.fn}
+        pagedShortLinks={pagedShortLinks}
+        pageSize={pageSize}
+      />
+    );
+
+    const expectedPagesCount = Math.ceil(pagedShortLinks.totalCount / pageSize);
+    expect(container.querySelectorAll('.page-number')).toHaveLength(
+      expectedPagesCount
+    );
+  });
+
+  test('should request short links for initial page correctly', () => {
+    const onPageLoad = jest.fn();
+
+    render(<UserShortLinksSection onPageLoad={onPageLoad} />);
+
+    expect(onPageLoad).toBeCalledTimes(1);
+  });
+
+  test('should request short links with non default page size when passed as prop', () => {
+    const onPageLoad = jest.fn();
+    const pageSize = 5;
+
+    render(
+      <UserShortLinksSection onPageLoad={onPageLoad} pageSize={pageSize} />
+    );
+
+    expect(onPageLoad).toBeCalledTimes(1);
+    expect(onPageLoad).toBeCalledWith(0, pageSize);
+  });
+
+  test('should request and render short links with correct offset when page changed', () => {
```

Comment:
Need to simplify this one. Ideally check state change instead of function calls. 
https://blog.pragmatists.com/test-doubles-fakes-mocks-and-stubs-1a7491dfa3da
https://martinfowler.com/articles/mocksArentStubs.html